### PR TITLE
[57423] Progress modal should always prevent save when there are invalid values

### PR DESCRIPTION
**Pull request by cbliard**
# Ticket

https://community.openproject.org/wp/57423

# What are you trying to accomplish?

In progress popover, when inputs are invalid, the modal can still be "saved" when the Save button is pressed multiple time. Nothing is actually changed, but this is still confusing behavior.

Change it so that the popover remains open.

## Screenshots

Before:

[clicking save twice dismisses the modal](https://github.com/user-attachments/assets/7a5e1289-8202-490f-84d0-c7e094109e63)

After:

[The modal with invalid values is still there even if save button is clicked multiple times](https://github.com/user-attachments/assets/c71371fb-5de6-4a9c-95d2-57954376bdf6)

# What approach did you choose and why?

The problem came from the modal form being updated through turbo-stream on submit. In this code path, the rendered modal form did not include the touched fields information, which lead to no fields being touched, meaning it saves without any modifications.

To fix it:
- include the touched information when rendering the modal

And to clarify the code:
- on `update` action:
  - on success, do not render any turbo-stream. The modal will be closed anyway.
  - on error, render turbo-stream directly from the controller instead of using the `update` template
- on `create` action:
  - on error, render turbo-stream directly from the controller instead of using the `update` template
- use `morph` instead of `update` with turbo-stream to avoid disconnecting and reconnecting stimulus controllers

And as a bonus:
- fixed an issue with the remaining work not being recalculated if status was changed on the new work package form.
- fix some test flakiness when checking for progress field focused state

# Merge checklist

- [x] Added/updated tests
- [ ] Added/updated documentation in Lookbook (patterns, previews, etc)
- [x] Tested major browsers (Chrome, Firefox, Edge, ...)

### CODE REVIEW

File: frontend/src/stimulus/controllers/dynamic/work-packages/progress/preview.controller.ts [hunk 1]
```diff
@@ -29,21 +29,82 @@
...
+        this.touchedFields.add(fieldName);
+      }
+    });
+
+    this.debouncedPreview = debounce((event:Event) => { void this.preview(event); }, 100);
+    // TODO: Ideally morphing in this single controller should not be necessary.
+    // Turbo supports morphing, by adding the <turbo-frame refresh="morph"> attribute.
+    // However, it has a bug, and it doesn't morphs when reloading the frame via javascript.
+    // See https://github.com/hotwired/turbo/issues/1161 . Once the issue is solved, we can remove
+    // this code and just use <turbo-frame refresh="morph"> instead.
```
Comment (oliverguenther) [inline frontend/src/stimulus/controllers/dynamic/work-packages/progress/preview.controller.ts:71]: Could it be that this is now fixed with 8.0.10? 

See:

https://github.com/hotwired/turbo/releases/tag/v8.0.10
https://github.com/hotwired/turbo/pull/1192

**Review by oliverguenther — APPROVED** [review]
👍 Nice cleanup. I feel like the update flow is now a lot easier. Current behavior fixed as specified in the expected behavior.

There is a small remark on whether the newest update from today might be able to resolve your workaround. I'm fine with merging it before checking.

File: frontend/src/stimulus/controllers/dynamic/work-packages/progress/preview.controller.ts (hunk 1 above)
Comment (cbliard) [reply frontend/src/stimulus/controllers/dynamic/work-packages/progress/preview.controller.ts:71]: Nice! I'll test that.

Comment (cbliard) [reply frontend/src/stimulus/controllers/dynamic/work-packages/progress/preview.controller.ts:71]: It seems I can't make it work properly. I'll merge this one, and open another PR.

**Pull request merged**